editor.planx.uk/src/@planx/components/Question/Public/Question.tsx

```diff
@@ -57,15 +59,17 @@ const Question: React.FC<IQuestion> = (props) => {
         a: previousResponseKey ?? undefined,
       },
     },
-    onSubmit: (values) => {
-      setTimeout(
-        () => props.handleSubmit({ answers: [values.selected.id] }),
-        theme.transitions.duration.standard,
-      );
-    },
-    validate: () => {
-      // do nothing
-    },
+    onSubmit: (values) => props.handleSubmit({ answers: [values.selected.id] }),
+    validateOnBlur: false,
+    validateOnChange: false,
+    validationSchema: object({
+      selected: object({
+        id: string().required("Select your answer before continuing"),
+        a: mixed().required().test(value => 
```

Comment:
`selected.a` can be `undefined` in `initialValues` - is it actually required here or optional? 

Also nice to see the `mixed()` syntax in action :muscle:

Reply:
Good question! I think what I have is correct, as this is a required value _at time of submission_. 

When the component loads the value is not set as it doesn't exist (so it's optional in the props). When navigating "back" it will be set already. The value for `selected.a` is set alongside `selected.id` here - https://github.com/theopensystemslab/planx-new/blob/61937317759cd41518d52a397a204584eb2e5a53/editor.planx.uk/src/%40planx/components/Question/Public/Question.tsx#L115-L116

Reply:
That distinction makes sense - thanks for clarifying!